File: app/src/test/java/no/nav/tilbakemeldingsmottak/itest/config/TestConfig.java
package no.nav.tilbakemeldingsmottak.itest.config;

import no.nav.security.token.support.spring.test.EnableMockOAuth2Server;
import no.nav.tilbakemeldingsmottak.CoreConfig;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;


@Configuration
@EnableAutoConfiguration
@EnableMockOAuth2Server
@Import({CoreConfig.class})
public class TestConfig {
}

File: app/src/test/kotlin/no/nav/tilbakemeldingsmottak/TestConfig.kt
package no.nav.tilbakemeldingsmottak

import no.nav.tilbakemeldingsmottak.consumer.email.aad.AADMailClient
import org.mockito.kotlin.mock
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

@TestConfiguration(proxyBeanMethods = false)
class TestConfig {

    @Bean
    @Primary
    fun aadMailClient(): AADMailClient = mock()
}
